### DISCUSSION TITLE
No, S0, S2 Security

### DISCUSSION THREAD

**Discussion by thesebastianf** [zwave-js/zwave-js#4886] [Q&A] [answered]
Hey,

out of my 70 nodes only 2 are paired with Security.
One is a lock and at the point in time when I paired it my former solution only allowed S0 Security (the lock would be capable of S2).
And one lightswitch is also paired with S0 ... I can't remember why for this one.

Out of all my nodes those are the 2 that have some hick ups in communication and are not really reliable.

Q:
- Is routing in zwave of S0 devices at all possible via non secure paired nodes or do those 2 devices maybe due to this have a bad performance?
- Is there a way in ZwaveJS to repair them with a different Security setting but keeping the same Node ID?

Thank you :)

**Comment by AlCalzone** [comment, marked as answer]
Encryption happens at the application layer, and does not influence routing at all. Encrypted messages are just arbitrary data for repeaters. Any mains-powered node can route any message, whether it is encrypted or not.

Bad performance with S0 can be caused by a suboptimal mesh, because with S0 every actual message requires 3 messages to be exchanged (get nonce, nonce report, encrypted message). S2 doesn't have that downside.

 > Is there a way in ZwaveJS to repair them with a different Security setting but keeping the same Node ID?(

With a trick and zwavejs2mqtt, yes:
* Shut down Z-Wave JS
* Factory reset the node in question
* Start Z-Wave JS again, the node should now be detected as dead (if mains-powered, doesn't matter for sleeping devices)
* Do a "Replace failed node" on the node, choose the desired security level. Once you're prompted for inclusion, put the node into inclusion mode. It should now be included again with the same node ID as earlier.

If you do an NVM backup beforehand, you can also try again in case this fails. Then you can restore the backup and repeat the replace operation.
